### PR TITLE
Introducing Linux support for the brofiler server

### PR DESCRIPTION
**Pull request by bfierz**
Includes the external work introducing linux support for the brofiler server

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+cmake_minimum_required (VERSION 2.8.11)
+project (Brofiler)
+
+IF (WIN32)
+add_definitions(-DWIN32)
+ELSEIF(WIN64)
+add_definitions(-DWIN64)
+ELSE()
+add_definitions(-DLINUX64)
```
Comment (bschindler) [inline CMakeLists.txt:9]: And linux32 does not exist?

File: ProfilerCore/Brofiler.h [hunk 2]
```diff
@@ -1,5 +1,13 @@
 #pragma once
 
+#include "Types.h"
+
+#ifdef LINUX64
+#include "Linux/PublicCommonLinux.h"
+#else
+#include "Windows/PublicCommonWindows.h"
```
Comment (bschindler) [inline ProfilerCore/Brofiler.h:8]: I think its much better to have a layout like: 

linux/os/PublicCommon.h
windows/os/PublicCommon.h

Depending on the os, you include linux or windows, and all you do is include <os/PublicCommon.h>

File: ProfilerCore/Core.cpp [hunk 3]
```diff
@@ -245,7 +245,7 @@ const std::vector<std::unique_ptr<ThreadEntry>>& Core::GetThreads() const
 	return threads;
 }
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
-EventStorage* Core::storage = nullptr;
+THREAD_LOCAL_VARIABLE EventStorage* Core::storage = nullptr;
```
Comment (bschindler) [inline ProfilerCore/Core.cpp:248]: How is this related to linux support?

File: ProfilerCore/Linux/SocketLinux.h [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+#include <cstring>
+#include <unistd.h>
+#include <sys/types.h> 
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <fcntl.h>
+
+namespace Profiler
+{
+	class Socket
```
Comment (bschindler) [inline ProfilerCore/Linux/SocketLinux.h:16]: Ugh... platform independent socket libraries do exist, you know...

File: CMakeLists.txt (hunk 1 above)
Comment (bfierz) [inline CMakeLists.txt:9]: Good point. We should test this

File: ProfilerCore/Brofiler.h (hunk 2 above)
Comment (bfierz) [inline ProfilerCore/Brofiler.h:8]: Just looked at the content. It seems to be more GCC/MSVC specific rather than OS specific

File: ProfilerCore/Core.cpp (hunk 3 above)
Comment (bfierz) [inline ProfilerCore/Core.cpp:248]: MSVC -> GCC compatibility. Not strictly linux.

File: ProfilerCore/Linux/SocketLinux.h (hunk 4 above)
Comment (bfierz) [inline ProfilerCore/Linux/SocketLinux.h:16]: Should we use asio (http://think-async.com/)?

**Comment by bfierz** [comment]
Should we merge this in this state, or try to refactor this branch first to have a proper handling different compilers and only then of different OSs?

**Comment by bschindler** [comment]
I think we should merge this as one block and then do new branches to individually fix this
